src/Estimators/OperatorEstBase.h

```diff
@@ -0,0 +1,163 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2020 QMCPACK developers.
+//
+// File developed by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
+//
+// File refactored from: OperatorBase.h
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+/**@file
+ */
+#ifndef QMCPLUSPLUS_OPERATORESTBASE_H
+#define QMCPLUSPLUS_OPERATORESTBASE_H
+
+#include <variant>
+
+#include "Particle/ParticleSet.h"
+#include "OhmmsData/RecordProperty.h"
+#include "Utilities/RandomGenerator.h"
+#include "QMCHamiltonians/observable_helper.h"
+#include "QMCWaveFunctions/OrbitalSetTraits.h"
+#include "type_traits/DataLocality.h"
+#include <bitset>
+
+namespace qmcplusplus
+{
+class DistanceTableData;
+class TrialWaveFunction;
+
+/** @ingroup Estimators
+ * @brief An abstract class for gridded estimators
+ *
+ */
+class OperatorEstBase
+{
+public:
+  using QMCT      = QMCTraits;
+  using MCPWalker = Walker<QMCTraits, PtclOnLatticeTraits>;
+
+  /** the type in this variant changes based on data locality
+   */
+  using Data = std::variant<std::unique_ptr<std::vector<QMCT::RealType>>, std::shared_ptr<std::vector<QMCT::RealType>>>;
+
+  ///enum to denote energy domain of operators
+  enum energy_domains
+  {
+    kinetic = 0,
+    potential,
+    no_energy_domain
+  };
+
+  enum quantum_domains
+  {
+    no_quantum_domain = 0,
+    classical,
+    quantum,
+    classical_classical,
+    quantum_classical,
+    quantum_quantum
+  };
+
+  /// locality for accumulation data
+  DataLocality data_locality_;
+
+  /// quantum_domain of the (particle) operator, default = no_quantum_domain
+  quantum_domains quantum_domain;
+  /// energy domain of the operator (kinetic/potential), default = no_energy_domain
+  energy_domains energy_domain;
+
+  /// enum for UpdateMode
+  enum
+  {
+    PRIMARY     = 0,
+    OPTIMIZABLE = 1,
+    RATIOUPDATE = 2,
+    PHYSICAL    = 3,
+    COLLECTABLE = 4,
+    NONLOCAL    = 5,
+  };
+
+  ///set the current update mode
+  std::bitset<8> UpdateMode;
+  ///current value
+  QMCT::FullPrecRealType Value;
+  ///a new value for a proposed move
+  QMCT::FullPrecRealType NewValue;
+  /// This is used to store the value for force on the source
+  /// ParticleSet.  It is accumulated if setComputeForces(true).
+  ParticleSet::ParticlePos_t IonForce;
+  //Walker<Return_t, ParticleSet::ParticleGradient_t>* tWalker;
+  ///name of this object
+  std::string myName;
+
+  QMCT::FullPrecRealType walkers_weight_;
+
+  QMCT::FullPrecRealType get_walkers_weight() const { return walkers_weight_; }
+  ///constructor
+  OperatorEstBase();
+  OperatorEstBase(const OperatorEstBase& oth);
+  ///virtual destructor
+  virtual ~OperatorEstBase() {}
```

Comment:
I think `virtual ~OperatorEstBase() = default;` is better than `virtual ~OperatorEstBase() {}`